zetaclient/tx.go

```diff
@@ -41,6 +41,11 @@ func (b *ZetaCoreBridge) WrapMessageWithAuthz(msg sdk.Msg) (sdk.Msg, AuthZSigner
 func (b *ZetaCoreBridge) PostGasPrice(chain common.Chain, gasPrice uint64, supply string, blockNum uint64) (string, error) {
 	signerAddress := b.keys.GetOperatorAddress().String()
 	msg := types.NewMsgGasPriceVoter(signerAddress, chain.ChainId, gasPrice, supply, blockNum)
+
+	if err := msg.ValidateBasic(); err != nil {
```

Comment:
Why not just call ValidateBasic in WrapMessageWithAuthz , and add an error to the return params if needed?
You could print `sdk.MsgTypeURL(msg)` into the error message

Reply:
Sounds good [15f22b4](https://github.com/zeta-chain/node/pull/1239/commits/15f22b4d8986088edf324314365fa80eeddfcfa6)